lib/v2/gdut/oa_news.js

```diff
@@ -0,0 +1,210 @@
+    },
+    notice: {
+        id: '5854888065150372255',
+        name: '校内通知',
+        publish: false,
+    },
+    announcement: {
+        id: '5821359576359193913',
+        name: '公示公告',
+        publish: false,
+    },
+    tender_result: {
+        id: '-1226046021292568614',
+        name: '招标结果',
+        publish: true,
+    },
+    tender_invite: {
+        id: '-3656117696093796045',
+        name: '招标公告',
+        publish: true,
+    },
+};
+
+function getArg(type) {
+    if (type.publish) {
+        return JSON.stringify([
+            {
+                pageSize: '20',
+                pageNo: 1,
+                listType: '1',
+                spaceType: '',
+                spaceId: '',
+                typeId: type.id,
+                condition: 'publishDepartment',
+                textfield1: '',
+                textfield2: '',
+                myNews: '',
+            },
+        ]);
+    } else {
+        return JSON.stringify([
+            {
+                pageSize: '20',
+                pageNo: 1,
+                listType: '1',
+                spaceType: '2',
+                spaceId: '',
+                typeId: '',
+                condition: 'publishDepartment',
+                textfield1: '',
+                textfield2: '',
+                myNews: '',
+                fragmentId: type.id,
+                ordinal: '0',
+                panelValue: 'designated_value',
+            },
+        ]);
+    }
+}
+
+module.exports = async (ctx) => {
+    const typeParam = ctx.params.type || 'notice';
+    if (typeMap[typeParam] === undefined) {
+        throw Error('通知类型' + typeParam + '未定义');
+    }
+
+    const type = typeMap[typeParam];
+
+    // 获取cookie
+    const cookieJar = new CookieJar();
+    await got({
+        method: 'get',
+        url: site + '/main.do',
+        cookieJar,
+    });
+
+    // 获取文章列表
+    const listUrl = '/ajax.do?method=ajaxAction&managerName=newsDataManager';
+    const resp = await got({
+        method: 'post',
+        url: site + listUrl,
+        cookieJar,
+        form: {
+            managerMethod: 'findListDatas',
+            arguments: getArg(type),
+        },
+    });
+
+    // 构造文章数组
+    const articles = [];
+    resp.data.list.forEach((item) => {
+        articles.push({
+            title: item.title,
+            guid: item.id,
+            link: site + '/newsData.do?method=newsView&newsId=' + item.id,
+            pubDate: new Date(item.publishDate1).toUTCString(),
+            author: item.publishUserDepart,
+            category: item.typeName,
+        });
+    });
+
+    // 获取实际的文章内容
+    await Promise.all(
+        articles.map(async (data) => {
+            const link = data.link;
+            data.description = await ctx.cache.tryGet(link, async () => {
+                // 获取数据
+                const response = await got({
+                    method: 'get',
+                    url: link,
+                    cookieJar,
+                });
```

Comment:
got request can be [simplified](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#json) to 
```suggestion
                const response = await got(link, {
                    cookieJar,
                });
```